http-netty/src/main/java/io/micronaut/http/netty/channel/NettyThreadFactoryConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.netty.channel;
+
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.bind.annotation.Bindable;
+import io.micronaut.core.util.StringUtils;
+
+/**
+ * Configuration for the default netty {@link java.util.concurrent.ThreadFactory}.
+ *
+ * @param reactorNonBlocking If {@code true} (the default), netty event loop threads will implement
+ *                           project reactor {@link reactor.core.scheduler.NonBlocking} by default.
+ *                           This will make reactor blocking operations throw an exception.
+ * @since 4.4.0
+ * @author Jonas Konrad
+ */
+@ConfigurationProperties("netty.default.thread-factory")
+@BootstrapContextCompatible
+@Internal
+public record NettyThreadFactoryConfiguration(
```

Comment:
do we need a whole new configuration properties class for this? Can it not go in `NettyGlobalConfiguration`?

Reply:
sure